### PR TITLE
Add handling for buggy PXE stacks.

### PR DESCRIPTION
**Pull request by jadestorm**
Hi folk!  This fixes an interesting situation that comes up with some PXE stacks and UEFI.  You can read about where this solution came from here:
http://rawtechnology.blogspot.com/2014/09/certain-machines-cant-boot-from-linux.html

Note: in my setup I don't have foreman handling dhcp.

I thought about adding support to add arbitrary lines to the tftpd.map file, but figured this might be a good global fix for anyone having these issues.

The client machines that had the buggy pxe stacks for me were hyper-v vms.

### CODE REVIEW

**Comment by ekohl** [comment]
I don't know enough about tftp to review the content change. Maybe @lzap?

For the puppet part it looks like rspec doesn't like these bytes. @domcleal any clue?

**Comment by jadestorm** [comment]
FWIW if y'all don't feel comfortable accepting it as is, I can go ahead with my original plan of adding in support to add arbitrary mappings to the file and I could configure my own to add the necessary line.  I suspect part of rspec's frustration is that it's latin1 charset rather than utf-8.  Since this potentially affects others though, I figured I'd give the quick route a shot.  ;)

**Comment by domcleal** [comment]
This looks fine to me. Regarding the UTF-8 error in the test, it's probably because the file content is not valid UTF-8 that it's unable to apply a regex against the string. I'd suggest moving the template to `files/` and the file resource from `content` to `source`.

File: manifests/config.pp [hunk 1]
```diff
@@ -29,9 +29,9 @@
       }
 
       file {'/etc/tftpd.map':
-        content => template('tftp/tftpd.map'),
-        mode    => '0644',
-        notify  => Class['xinetd'],
+        source => 'puppet:///modules/tftp/tftpd.map',
```
Comment (ekohl) [inline manifests/config.pp]: I believe that for small files it's faster to use ```content => file('tftp/tftpd.map')```.

Comment (jadestorm) [inline manifests/config.pp]: Fine with me either way =)  Is that what y'all would prefer?

**Comment by jadestorm** [comment]
@domcleal Switched to a file instead.  Thought @ekohl has indicated I should switch it back to a content with a file reference.  Either method is fine with me so I'll go with a "just let me know which way y'all want it and I'll make it happen".

**Comment by jadestorm** [comment]
Am I causing these check failures btw?  They don't look related to my patch.

**Comment by ekohl** [comment]
They are. They verify that the file ```tftpd.map``` is present with a property ```content``` that at least contains a text. Since you moved that to source you either need to update the tests or implement my suggestion.

**Comment by jadestorm** [comment]
Ah!  Ok then  =)  I'll make that change tomorrow!

**Comment by jadestorm** [comment]
@ekohl Ok I switched it, but the tests are still mad about the non-UTF-8 character.  Hrm.  I'm afraid I'm not fluent enough with rspec and such to know if there's a way to say "this is Latin-1, treat it as such".  If this is causing too much trouble though I can always go back to my older idea, which was to add some very simplistic handling of additional mappings that the template would include.  Thoughts?

**Comment by domcleal** [comment]
Yeah, I don't think using `content => file()` is a good idea because of the invalid UTF-8 content in the file. Using `source` to access the file is probably the best way to handle the binary content, then change the spec to use `with_source` rather than content. It'll mean the content of the file can't be tested, but that's OK.

File: manifests/config.pp [hunk 2]
```diff
@@ -29,7 +29,7 @@
       }
 
       file {'/etc/tftpd.map':
-        content => template('tftp/tftpd.map'),
+        source  => 'puppet:///modules/tftp/tftpd.map',
```
Comment (mmoll) [inline manifests/config.pp]: Should this be changed to `${module_name}`, while somebody is at it, anyway?

File: manifests/config.pp [hunk 3]
```diff
@@ -29,7 +29,7 @@
...
         mode    => '0644',
         notify  => Class['xinetd'],
```
Comment (mmoll) [inline manifests/config.pp]: there's too much space before the arrow now.

File: manifests/config.pp (hunk 2 above)
Comment (jadestorm) [inline manifests/config.pp]: Interesting, I wasn't aware that was a thing!

File: manifests/config.pp (hunk 3 above)
Comment (jadestorm) [inline manifests/config.pp]: <_<  Yeah I am trying to do too many things at the same time and didn't run my typical lint-via-vim.  Dooh!

**Comment by domcleal** [comment]
Merged, thanks @jadestorm.

**Pull request merged**